tomopy/prep/alignment.py

```diff
@@ -443,3 +455,240 @@ def shift_images(prj, sx, sy):
     prj *= scl
 
     return prj
+
+
+def find_slits_corners_aps_1id(img, 
+                               method='quadrant+', 
+                               medfilt2_kernel_size=3,
+                               medfilt_kernel_size=23,
+                               ):
+    """
+    Automatically locate the slit box location by its four corners.
+
+    Parameters
+    ----------
+    img                   :  np.2darray
+        input 2D images
+    method                :  str,  ['simple', 'quadrant', 'quadrant+'], optional
+        method for auto detecting slit corners
+    medfilt2_kernel_size  :  int, optional 
+        2D median filter kernel size for noise reduction
+    medfilt_kernel_size   :  int, optional 
+        1D median filter kernel size for noise reduction
+    
+    Returns
+    -------
+    tuple
+        autodetected slit corners (counter-clockwise order)
+        (upperLeft, lowerLeft, lowerRight, upperRight)
+    """
+
+
+    img = medfilt2d(np.log(img.astype(np.float64)),
+                    kernel_size=medfilt2_kernel_size,
+                    )
+    rows, cols = img.shape
+    
+    # simple method is simple, therefore it stands out
+    if method.lower() == 'simple':        
+        # assuming a rectangle type slit box 
+        col_std = medfilt(np.std(img, axis=0), kernel_size=medfilt_kernel_size)
+        row_std = medfilt(np.std(img, axis=1), kernel_size=medfilt_kernel_size)
+        # NOTE: in the tiff img
+        #  x is col index, y is the row index  ==> key point here !!!
+        #  img slicing is doen with img[row_idx, col_idx]
+        #  ==> so the image idx and corner position are FLIPPED!
+        _left   = np.argmax(np.gradient(col_std))
+        _right  = np.argmin(np.gradient(col_std))
+        _top    = np.argmax(np.gradient(row_std))
+        _bottom = np.argmin(np.gradient(row_std))
+
+        cnrs = np.array([[_left , _top    ],
+                         [_left , _bottom ],
+                         [_right, _bottom ],
+                         [_right, _top    ],
+                        ])
+    else:
+        # predefine all quadrants
+        # Here let's assume that the four corners of the slit box are in the 
+        # four quadrant defined by the center of the image
+        # i.e.
+        #  uppper left quadrant: img[0     :cnt[1], 0     :cnt[0]]  => quadarnt origin =  (0,           0)
+        #  lower  left quadrant: img[cnt[1]:      , 0     :cnt[0]]  => quadarnt origin =  (cnt[0],      0)
+        #  lower right quadrant: img[cnt[1]:      , cnt[0]:      ]  => quadarnt origin =  (cnt[0], cnt[1])
+        #  upper right quadrant: img[0     :cnt[1], cnt[0]:      ]  => quadarnt origin =  (0,      cnt[1])
+        cnt = [int(cols/2), int(rows/2)]  # center of image that defines FOUR quadrants
+        Quadrant  = namedtuple('Quadrant', 'img col_func, row_func')

```

Comment:
vars should be all lowercase

Reply:
I use the namedtuple() here as a shorthand for an in-line class.